scripts/clean-shrinkwrap.js

```diff
@@ -0,0 +1,123 @@
+ * @param {*} depType - Dependency type (dependencies|requires)
+ *
+ * @example
+ * removeDependency(keys, tree, onNode, parentName)
+ */
+const removeDependency = (keys, tree, onNode, parentName, depType) => {
+  let name = null
+  for (let index = 0; index < keys.length; index += 1) {
+    name = keys[index]
+
+    // Check for this dependency to still exist,
+    // as a node might have been removed just before this iteration
+    if (tree[depType][name]) {
+      onNode(tree, parentName || tree.name, name, tree[depType][name], depType)
+
+      // Check that the walking function didn't remove the dependency
+      // if so, skip it and continue with the next one
+      if (tree[depType][name] && tree[depType][name][depType]) {
+        traverseDeps(tree[depType][name], onNode, name)
+      }
+    }
+  }
+}
+
+/**
+ * @summary Traverse a shrinkwrap tree and call
+ * a given function on each dependency
+ * @param {Object} tree - shrinkwrap
+ * @param {Function} onNode - callback({Object} parent, {String} parentName, {String} name, {Object} info)
+ * @param {String} [parentName] - name of dependent (used internally)
+ * @example
+ * traverseDeps(shrinkwrap, (parent, parentName, name) => {
+ *   // ...
+ * })
+ */
+const traverseDeps = (tree, onNode, parentName) => {
+  if (!tree.dependencies && !tree.requires) {
+    return
+  }
+
+  const depsKeys = Object.keys(tree.dependencies)
+  removeDependency(depsKeys, tree, onNode, parentName, 'dependencies')
+
+  if (_.isObject(tree.requires)) {
+    const reqsKeys = Object.keys(tree.requires)
+    removeDependency(reqsKeys, tree, onNode, parentName, 'requires')
+  }
+}
+
+console.log('Cleaning shrinkwrap...')
+
+// Walk the generated shrinkwrap tree & apply modifications if necessary
+traverseDeps(shrinkwrap, (parent, parentName, name) => {
+  // If this dependency depends on a "blacklisted" optional
+  // dependency; remove it from the shrinkwrap
+  if (packageInfo.platformSpecificDependencies.includes(name)) {
+    console.log(' - Removing "%s" from "%s" dependencies', name, parentName)
+    parent.dependencies[name] = undefined
+    delete parent.dependencies[name]
+
+    // Check that we're not in the root object
+    // before removing the "requires" property
+    if (_.isObject(parent.requires)) {
+      console.log(' - Removing "%s" from "%s" requires', name, parentName)
+      parent.requires[name] = undefined
+      delete parent.requires[name]
+    }
+  }
+})
+
+// Generate the new shrinkwrap JSON
+const shrinkwrapJson = `${JSON.stringify(shrinkwrap, null, JSON_INDENT)}\n`
+
+// Write back the modified npm-shrinkwrap.json
+fs.writeFile(SHRINKWRAP_FILENAME, shrinkwrapJson, (error) => {
+  if (error) {
+    console.log(`[ERROR] Couldn't write shrinkwrap file: ${error.stack}`)
+    process.exitCode = 1
+    process.exit()
+  } else {
+    console.log(`[OK] Wrote shrinkwrap file to ${path.relative(__dirname, SHRINKWRAP_FILENAME)}`)
+  }
+})
+
+console.log('')
```

Comment:
This whole file can be replaced with something like 
```js
const fs = require('fs');
const omit = require('omit-deep-lodash');
const shrinkwrap = require('./npm-shrinkwrap.json')
const cleaned = omit(shrinkwrap, 'fsevents', 'winusb-driver-generator');
fs.writeFileSync('./npm-shrinkwrap.json', JSON.stringify(cleaned, null, 2));
```

Reply:
I'm not sure `omit` works by removing deep properties as well like dependencies of dependencies, but it can be solved with a cycle/recursive function. Trying now

Reply:
That's exactly what ``omit-seep-lodash does, see https://www.npmjs.com/package/omit-deep-lodash .

Reply:
Didn't notice, nice!